gym_snake/envs/snakeGameGym.py

```diff
@@ -179,6 +184,27 @@ def check_body_collision(self) -> bool:
 
 		return False
 
+	def check_exceeded_max_moves(self) -> bool:
+		"""
+		Function that checks whether or not the snake has made too many moves since 
+		last consuming a fruit.
+		"""
+		return (self.max_moves_no_fruit <= 0 or # non-positive allowed move count default returns true
+				self.check_fruit_collision() or # if the snake's head is in the same space as a fruit return true
+				self.num_moves_since_fruit <= self.max_moves_no_fruit) #
```

Comment:
I think the conditionals might be opposite here. if `self.num_moves_since_fruit <= self.max_moves_no_fruit` is True, shouldn't we want `check_exceeded_max_moves` to return False? Same for `self.check_fruit_collision()` if this is True, I think we want `check_exceeded_max_moves` to return False.

OR

if this conditional is working as intended maybe the issue is the name of the function should be flipped? Like `check_under_max_moves` or something?

Reply:
Yup you're totally right. Before changing up the naming (which is totally cleaner with your suggestions), that method was named `check_num_moves_since_fruit_valid` and would return `True` if the snake was still within the allowable move threshold.

Now the name has been changed to `check_exceeded_max_moves` and the logic has been updated to reflect it, i.e. it returns `False` if the snake is still within the allowable move threshold.